src/physics/cam/nn_interface_cam.F90

```diff
@@ -593,28 +603,30 @@ subroutine CAM_var_conversion(qv, qc, qi, q, tabs, t)
             !! Absolute temperature from CAM
 
 
-        nx = size(q, 1)
-        nz = size(q, 2)
+        nx = size(r, 1)
+        nz = size(r, 2)
 
         do k = 1, nz
           do i = 1, nx
-            q(i,k) = qv(i,k) + qc(i,k) + qi(i,k)
+            ! Calculate dry mixing ratio as required by SAM from moist variables as provided by CAM
+            rv = qv(i,k) / (1. - qv(i,k))
+            r(i,k) = rv + qc(i,k)*(1.+rv) + qi(i,k)*(1.+rv)
 
             ! omp  = max(0.,min(1.,(tabs(i,k)-tprmin)*a_pr))
             omn  = max(0.,min(1.,(tabs(i,k)-tbgmin)*a_bg))
             t(i,k) = tabs(i,k) &
                    ! - (fac_cond+(1.-omp)*fac_fus)*qp(i,k) &  ! There is no qp in CAM
-                     - (fac_cond+(1.-omn)*fac_fus)*(qc(i,k) + qi(i,k)) &
+                     - (fac_cond+(1.-omn)*fac_fus)*(qc(i,k)*(1.+rv) + qi(i,k)*(1.+rv)) &
                      + gamaz(k)
           end do
         end do
 
     end subroutine CAM_var_conversion
     
```

Comment:
Move conversion to rc and ri out of ll. 619 to be below 613, and create an explicit ri and rc variable.